### PR TITLE
Issue 29213 Copying a contentlet without having Edit Permissions causes copy to have incorrect permissions.

### LINKED ISSUE #29213: Copying a contentlet without having Edit Permissions causes copy to have incorrect permissions.
### Parent Issue

_No response_

### Problem Statement

When a user who has full View, Edit and Publish permissions on a content type, but not Edit Permissions goes to copy a contentlet of that content type the copy goes through but does not apply the permissions of the original. This also throws an error in the log saying  "User id: XXXXXX does not have permission to alter permissions on asset XXXXXX"

The customer in question experiencing it does not see the contentlet they have copied as the permissions on the new contentlet do not match the users. 

### Steps to Reproduce

On the demo site I have created a content type called "Permissions Test" which has View, Edit and Publish permissions for the Back-End User Role. 

1. Create a user with the Back-End user role and any other required permissions to allow them to see the demo site. 
2. Create a contentlet of the "Permissions Test" content type. 
3. Open an additional tab to run the logs
4. Create a copy of the contentlet created to see the error. 

### Acceptance Criteria

Proposed by John one of the two following. 
1. When running the Copy Workflow, copy the permissions of the original content regardless of the current user's rights on Editting Permissions is
2. If the user has Edit Permissions rights then copy the original permissions. Otherwise, create new permissions on the new contentlet based on the current user's permissions, as if they have just created a new content item instead of copying one. 

### dotCMS Version

23.10

### Proposed Objective

User Experience

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.zendesk.com/agent/tickets/117180

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by gortiz-dotcms**
When executing a copy, a new boolean property IS_COPY is added to the contentlet, so this way if the contentlet is being copied, the validation that checks if the user has the edit permissions is skipped. This is the approach number 1 proposed in the issue.

Tests added

This PR fixes: #29213

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 1]
```diff
@@ -262,7 +262,7 @@ public class ESContentletAPIImpl implements ContentletAPI {
 
     public static final String UNIQUE_PER_SITE_FIELD_VARIABLE_NAME = "uniquePerSite";
 
-
+    private final String IS_COPY = "isCopy";
```
Comment (erickgonzalez) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java:265]: Looks like we already have a content property https://github.com/dotCMS/core/blob/4c3a898a308cadc2247a06fbee5397ed629d04f8/dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java#L170 let's check if we can use it. If not let's add the property to the class mentioned, so you can call it from diff places.

**Review by erickgonzalez — CHANGES_REQUESTED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java [hunk 2]
```diff
@@ -655,10 +655,18 @@ public void save(Permission permission, Permissionable permissionable, User user
 	 */
 	@WrapInTransaction
 	private void save(Permission permission, Permissionable permissionable, User user, boolean respectFrontendRoles, boolean createEvent) throws DotDataException, DotSecurityException {
-		if(!doesUserHavePermission(permissionable, PermissionAPI.PERMISSION_EDIT_PERMISSIONS, user)) {
+		boolean isCopy = false;
+		if (permissionable instanceof Contentlet){
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java:659]: I would encapsulate this into a method isCopy()
this methods are already so big

File: dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java [hunk 3]
```diff
@@ -655,10 +655,18 @@ public void save(Permission permission, Permissionable permissionable, User user
...
 	@WrapInTransaction
 	private void save(Permission permission, Permissionable permissionable, User user, boolean respectFrontendRoles, boolean createEvent) throws DotDataException, DotSecurityException {
-		if(!doesUserHavePermission(permissionable, PermissionAPI.PERMISSION_EDIT_PERMISSIONS, user)) {
+		boolean isCopy = false;
+		if (permissionable instanceof Contentlet){
+			Contentlet contentlet = (Contentlet) permissionable;
+			isCopy = contentlet.getBoolProperty(Contentlet.IS_COPY);
+		}
+
+		if(!isCopy) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java:664]: the same herem checkCopyPermissions or so

**Review by jdotcms — APPROVED** [review]
I have left some optional comments just in case

**Pull request merged**